lib/semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,341 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/parser/message.h"
+#include "flang/semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages) : messages_{messages} {}
+
+  void ReportOverlapCaseValError(
+      parser::CharBlock &source1, parser::CharBlock &source2) {
+    // C1149
+    messages_
+        .Say(source2,
+            "SELECT CASE statement value at this location overlaps with below location"_err_en_US)
+        .Attach(source1, ""_err_en_US);
```

Comment:
I don't think an empty message is a good idea. Why not specify the value that occurs in both CASE statements?

Reply:
This is how the error looks:
test4.f90:18:7: error: SELECT CASE statement value at this location overlaps with below location
        case (81:90)
        ^^^^^^^^^^^^
test4.f90:13:7:
        case (90)
        ^^^^^^^^^
The location and the values are already part of this error message. So I left it blank.

Reply:
Do not emit a blank message.  Add some descriptive text.

Reply:
Thank you. I will add similar message.